File: routes/GET/getList.py
from classes import Lists
from flask import make_response, jsonify

def get(profileId):
    try:
        list = Lists.List(profileId=profileId)
        getAllLists = list.getAll() 
        match getAllLists["message"]:
            case "The given profile does not exist":
                resp = make_response(jsonify(getAllLists))
                resp.status_code = 404
                return resp
            case "Lists found":
                resp = make_response(jsonify(getAllLists))
                resp.status_code = 200
                return resp
            case "An error has occurred while getting the list":
                resp = make_response(jsonify(getAllLists))
                resp.status_code = 500
                return resp
    except Exception as e:
        resp = make_response(jsonify({"message":"There was an error"}))
        resp.status_code = 500
        return resp

File: src/routes/GET/getList.ts
import List from "../../classes/lists";
import type { Response, Request } from "express";

export default async function getLists(req: Request, res: Response) {
  try {
    const profileId = req.params.profileId;
    const allLists = await new List(0, "", profileId).getAll();
    switch (allLists.message) {
      case "Profile not found":
        res.statusCode = 404;
        res.json(allLists);
        break;
      case "Success":
        res.statusCode = 200;
        res.json(allLists);
        break;
      case "An error occurred":
        res.statusCode = 500;
        res.json(allLists);
    }
  } catch (error: any) {
    res.statusCode = 500;
    res.json({ message: "There was an error", error: error.message });
  }
}
